src/ngraph/op/sigmoid.cpp

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph/op/sigmoid.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+shared_ptr<Node> op::Sigmoid::copy_with_new_args(const NodeVector& new_args) const
+{
+    if (new_args.size() != 1)
+    {
+        throw ngraph_error("Incorrect number of new arguments");
+    }
+
+    return make_shared<Sigmoid>(new_args.at(0));
+}
+
+op::Sigmoid::Sigmoid(shared_ptr<Node> arg)
+    : UnaryElementwiseArithmetic("Sigmoid", {arg})
+{
+    set_value_type_checked(arg->get_element_type(), arg->get_shape());
```

Comment:
It won't hurt anything, but this call is not necessary. (`UnaryElementwise`'s constructor does it for you.)